src/algebra/lie/free.lean

```diff
@@ -225,4 +220,18 @@ by { rw ← lift_symm_apply, exact (lift R).apply_symm_apply F, }
 have h' : (lift R).symm F₁ = (lift R).symm F₂, { ext, simp [h], },
 (lift R).symm.injective h'
 
+variables (R X)
+
+/-- The universal enveloping algebra of the free Lie algebra is just the free unital associative
+algebra. -/
+def universal_enveloping_equiv_free_algebra :
+  universal_enveloping_algebra R (free_lie_algebra R X) ≃ₐ[R] free_algebra R X :=
+{ to_fun    := universal_enveloping_algebra.lift R $ free_lie_algebra.lift R $ free_algebra.ι R,
+  inv_fun   := free_algebra.lift R $ (universal_enveloping_algebra.ι R) ∘ (free_lie_algebra.of R),
+  left_inv  := show ∀ x, _ = alg_hom.id R _ x,
+    by { simp only [← alg_hom.comp_apply, ← alg_hom.ext_iff], ext, simp, },
+  right_inv := show ∀ x, _ = alg_hom.id R _ x,
+    by { simp only [← alg_hom.comp_apply, ← alg_hom.ext_iff], ext, simp, },
```

Comment:
I think this proof can be made shorter with `alg_equiv.of_alg_hom`, which is how I usually construct these equivs.

Reply:
Thanks, `alg_equiv.of_alg_hom` is exactly the right tool for this job. I had to insert some type annotations to get it to work but arguably, these make it clearer anyway.